src/dbnode/persist/fs/msgpack/decoder.go

```diff
@@ -425,15 +475,60 @@ func (dec *Decoder) decodeIndexEntry(bytesPool pool.BytesPool) schema.IndexEntry
 	actualChecksum := dec.readerWithDigest.digest().Sum32()
 
 	// Decode checksum field originally added in V3
-	expectedChecksum := uint32(dec.decodeVarint())
-
-	if expectedChecksum != actualChecksum {
+	v := dec.decodeVarint()
+	indexEntry.IndexChecksum = uint32(v)
+	if indexEntry.IndexChecksum != actualChecksum {
 		dec.err = errorIndexEntryChecksumMismatch
 	}
 
 	return indexEntry
 }
 
+func (dec *Decoder) decodeIndexChecksum(
+	compareID []byte,
+	bytesPool pool.BytesPool,
+) (int64, IndexChecksumLookupStatus) {
+	numFieldsToSkip, actual, ok := dec.checkNumIndexFields()
+	if !ok || dec.err != nil {
+		return 0, NotFound
+	}
+
+	if actual-numFieldsToSkip < 7 {
```

Comment:
nit: maybe a comment explaining why `7` here.

Reply:
Actually is there a more descriptive way of doing this? I want to ensure the file version is at least the latest version (although actually now that I'm not using `IndexChecksum` I'm considering just dropping this restriction)

Reply:
Ah, so given that, the first thing is that you want to drop `-numFieldsToSkip`. You can just check if `actual` < 7. 

That aside, what you're doing is what we've typically been doing, but what we probably want is some helper method that just wraps this check and makes it a little more descriptive.

Reply:
Yeah damn, that's a little annoying... I'll probably just go ahead and drop the requirement for now since we instead compute the checksum manually